torch/csrc/profiler/util.cpp

```diff
@@ -788,4 +909,54 @@ uint64_t computeFlops(
   return 0;
 }
 
+// A function that takes an IValue
+// and returns a conventional string representation of the IValue
+// Currently it returns int representation of the last 20 bits of the address
+// value
+int getTensorStartHint(const at::Tensor& t) {
+  const auto tensor_impl = t.unsafeGetTensorImpl();
+  uintptr_t storage_addr = 0;
+  storage_addr = reinterpret_cast<uintptr_t>(tensor_impl->storage().data());
+  int last_bits = static_cast<int>(storage_addr & 0xFFFFF);
+  return last_bits;
+}
+
+bool checkFunctionOutputsForLogging(
+    const at::RecordFunction& fn,
+    const char* fn_name) {
+  const auto& outputs = fn.outputs();
+  auto num_outputs = fn.num_outputs();
+  VLOG(2) << "outputs: " << num_outputs << " " << outputs.size() << '\n';
+  // We have two cases: for unboxed kernel, we have num_outputs ==
+  // outputs.size() for boxed kernel using stack, there could be more elements
+  // on the stack from previous ops.
+  // TORCH_INTERNAL_ASSERT(num_outputs <= outputs.size());
+  if (num_outputs > outputs.size()) {
+    // LOG(WARNING) << "Profiler: RecordFunction " << fn_name
+    //              << " expected num_outputs=" << num_outputs
+    //              << " > outputs.size()=" << outputs.size();
+    return false;
+  }
+  return true;
+}
+
+bool checkFunctionInputsForLogging(
+    const at::RecordFunction& fn,
+    const char* fn_name) {
+  auto num_inputs = fn.num_inputs();
+  const auto inputs = fn.inputs();
+  VLOG(2) << "inputs: " << num_inputs << " " << inputs.size() << '\n';
+  // We have two cases: for unboxed kernel, we have num_inputs ==
+  // inputs.size() for boxed kernel using stack, there could be more elements
+  // on the stack from previous ops.
+  // TORCH_INTERNAL_ASSERT(num_inputs <= inputs.size());
+  if (num_inputs > inputs.size()) {
+    // LOG(WARNING) << "RecordFunction " << fn_name
```

Comment:
remove?

Reply:
I can remove but I also did not want to lost information on what it means when num_inputs > inputs.size(). Maybe I will just convert it to a comment